inc/styles-admin.php

```diff
@@ -282,6 +282,17 @@ class="image-fallback widefat" />
 				<?php
 				break;
 
+			case 'radio' :
+				foreach ( $field['options'] as $k => $v ) : ?>
+					<label for="<?php echo esc_attr( $field_name . '-' . $k ) ?>">
```

Comment:
I briefly looked into this recently and it quickly became more effort than it's worth, seeing as a `radio` input could just be replaced with a `select` input. The problem with radio inputs is that they need to have unique names and this will generate the same names for each Page Builder row, and then the browser thinks they're all part of the same radio input group. Check what happens when you attempt to use this and click the radio input labels for different Page Builder rows.

You could try generating a unique id to append/prepend to each of these, but again, IMO it's not worth spending a lot of time on this as `select` inputs are a perfectly good substitute.

Reply:
Hi, 
Radio buttons can't always be replaced by a select input.
Let's take this usecase for example:
If I want to give user a simple palette of theme defined colors, I can use and style the radio buttons:
For such use case radio buttons are much more appropriate then select, as all the possibilities are shown to the user.
Here is a simple demo:
https://jmp.sh/VKvDxd2

I could add the colors to the IRIS palete color picker, but if theme color changes later, then the user would need to chage it everywhere all over again. Now simply the background class is added to the cell and the color will always match the theme color.

For the technical part of unique namings, you can consider something like this:

```
<?php
case 'radio' :
	$radio_unique_id = uniqid() . '-';
	foreach ( $field['options'] as $k => $v ) : ?>
		<label for="<?php echo esc_attr( $field_name . '-' . $k ) ?>">
			<input type="radio" name="<?php echo esc_attr($radio_unique_id) . esc_attr( $field_name ) ?>"
				id="<?php echo esc_attr( $field_name . '-' . $k ) ?>" class="siteorigin-widget-input"
				value="<?php echo esc_attr( $k ) ?>" <?php checked( $k, $current ) ?>> <?php echo esc_html( $v ) ?>
		</label>
		<?php
	endforeach;
	break;
```

Reply:
How does that look? Not too sure how these fields would get reliably sanitized though.